leveldb/leveldb.go

```
@@ -0,0 +1,101 @@
+package leveldb
+
+import (
+	"fmt"
+
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/opt"
+)
+
+// Manager struct
+type Manager struct {
+	db *leveldb.DB
+}
+
+// Item struct
+type Item struct {
+	Key []byte
+	Value []byte
+}
+
+// IManager - Manager interface
+type IManager interface {
+	Close()
+	GetAllItems() (items []Item, err error)
+	AddNewItem(item Item) (status int, err error)
+	FindItemByKey(key []byte) (item *Item, err error) 
+}
+
+// Error - custom leveldb error struct
+type Error struct {
+	Key []byte
+	err error
+}
+
+// NewErr - create new custom error instance
+func NewErr(key []byte, err error) error {
```

Shouldn't customize the error, just need to throw the error from lower level.